clients/v1_auth/v1_auth.go

```diff
@@ -0,0 +1,428 @@
+	if &authPtr != nil {
+		return fmt.Errorf("authorization with username %q already exists", params.Username)
+	}
+	if err != nil {
+		return fmt.Errorf("failed to verify username %q has no auth: %w", params.Username, err)
+	}
+
+	password := params.Password
+	if password == "" && !params.NoPassword {
+		password, err = c.CLI.StdIO.GetSecret("Please enter your password", 0)
+		if err != nil {
+			return err
+		}
+	}
+
+	bucketPerms := []struct {
+		action string
+		perms  []string
+	}{
+		{action: "read", perms: params.ReadBucket},
+		{action: "write", perms: params.WriteBucket},
+	}
+
+	var permissions []api.Permission
+	for _, bp := range bucketPerms {
+		for _, p := range bp.perms {
+			// verify the input ID
+			if _, err := influxid.IDFromString(p); err != nil {
+				return fmt.Errorf("invalid bucket ID '%s': %w (did you pass a bucket name instead of an ID?)", p, err)
+			}
+
+			newPerm := api.Permission{
+				Action: bp.action,
+				Resource: api.PermissionResource{
+					Type:  "buckets",
+					Id:    *api.NewNullableString(&p),
+					OrgID: *api.NewNullableString(&orgID),
+				},
+			}
+			permissions = append(permissions, newPerm)
+		}
+	}
+
+	authReq := &api.AuthorizationPostRequest{
+		Description: &params.Desc,
+		OrgID:       orgID,
+		Permissions: permissions,
+		Token:       &params.Username,
+	}
+
+	// TODO Something wrong here
+	newAuth, err := c.AuthorizationsApi.PostAuthorizations(ctx).AuthorizationPostRequest(authReq).Execute()
+	if err != nil {
+		return fmt.Errorf("failed to create new authorization: %w", err)
+	}
+
+	if password != "" {
+		if err := c.UsersApi.PostUsersIDPassword(ctx, newAuth.GetUserID()).
+			PasswordResetBody(api.PasswordResetBody{
+				Password: password,
+			}).Execute(); err != nil {
+			_ = c.AuthorizationsApi.DeleteAuthorizationsID(ctx, newAuth.GetId()).Execute()
+			return fmt.Errorf("failed to update password for user ID %q: %w", newAuth.GetUserID(), err)
+		}
+	}
+
+	usr, err := c.UsersApi.GetUsersID(ctx, newAuth.GetUserID()).Execute()
+	if err != nil {
+		return err
+	}
+
+	ps := make([]string, 0, len(auth.Permissions))
+	for _, p := range auth.Permissions {
+		ps = append(ps, permString(p))
+	}
+
+	return c.printV1Tokens(&v1PrintOpts{
+		token: &v1Token{
+			ID:          newAuth.GetId(),
+			Description: newAuth.GetDescription(),
+			Token:       newAuth.GetToken(),
+			Status:      newAuth.GetStatus(),
+			UserName:    usr.GetName(),
+			UserID:      usr.GetId(),
+			Permissions: ps,
+		},
+	})
+}
+
+type RemoveParams struct {
+	clients.AuthLookupParams
+}
+
+func (c Client) Remove(ctx context.Context, params *RemoveParams) error {
+	id, err := c.getAuthReqID(ctx, &params.AuthLookupParams)
+	if err != nil {
+		return err
+	}
+
+	auth, err1 := c.AuthorizationsApi.GetAuthorizationsID(ctx, id).Execute()
+	err2 := c.AuthorizationsApi.DeleteAuthorizationsID(ctx, id).Execute()
+	if err1 != nil || err2 != nil {
+		return fmt.Errorf("could not find Authorization from ID %q: %w", id, err1)
```

Comment:
This is the error to show if `err1 != nil`, we should have a different error if `err2 != nil`